samples/batch/transcription-enabled-storage/CreateByTimerFunction/CreateTranscriptionByTimer.cs

```diff
@@ -0,0 +1,348 @@
+                    audioFileUrls.Add(new Uri(audioFileUrl));
+                }
+
+                if (CreateTranscriptionEnvironmentVariables.CreateReceiptFile)
+                {
+                    foreach (var audioFileUrl in audioFileUrls)
+                    {
+                        var audioFileName = StorageUtilities.GetFileNameFromPath(audioFileUrl.AbsoluteUri);
+
+                        var byteArray = await StorageUtilities.DownloadFileFromSAS(audioFileUrl.AbsoluteUri).ConfigureAwait(false);
+                        var audioDetails = StorageUtilities.GetAudioDetails(
+                            byteArray,
+                            audioFileName,
+                            locale,
+                            StartDateTime,
+                            modelIds.Any(),
+                            CreateTranscriptionEnvironmentVariables.AddSentimentAnalysis,
+                            CreateTranscriptionEnvironmentVariables.AddEntityRedaction,
+                            log);
+
+                        await StorageUtilities.WriteTranscriptionReceiptToStorageAsync(CreateTranscriptionEnvironmentVariables.AzureWebJobsStorage, CreateTranscriptionEnvironmentVariables.ReceiptsContainer, audioDetails, log).ConfigureAwait(false);
+                    }
+                }
+
+                var client = new BatchClient(subscription.SubscriptionKey, subscription.LocationUri.AbsoluteUri, log);
+                var transcriptionLocation = await client.PostTranscriptionAsync(
+                    jobName,
+                    "CreateByTimerTranscription",
+                    locale,
+                    properties,
+                    audioFileUrls,
+                    modelIds).ConfigureAwait(false);
+
+                log.LogInformation($"Location: {transcriptionLocation}");
+                locationString = transcriptionLocation.ToString();
+            }
+            catch (WebException e)
+            {
+                if (BatchClient.IsThrottledOrTimeoutStatusCode(((HttpWebResponse)e.Response).StatusCode))
+                {
+                    log.LogError($"Throttled or timeout while creating post, re-enqueueing transcription create. Message: {e.Message}");
+
+                    var createTranscriptionSBConnectionString = CreateTranscriptionEnvironmentVariables.CreateTranscriptionServiceBusConnectionString;
+                    log.LogInformation($"CreateTranscriptionSBConnectionString from settings: {createTranscriptionSBConnectionString}");
+
+                    foreach (var message in messages)
+                    {
+                        ServiceBusUtilities.SendServiceBusMessageAsync(createTranscriptionSBConnectionString, message, log, 2).GetAwaiter().GetResult();
+                    }
+
+                    return string.Empty;
```

Comment:
What happens for those messages? retry will be done at a later time?

Reply:
exactly. the idea here is that if we get throttled or get a timeout, we will retry sending the same message with a delay instead of failing the job